lib/options.ts

```diff
@@ -21,6 +21,35 @@ export class Options {
 
 	public options: IDictionary<IDashedOption>;
 
+	public setupOptions(projectData: IProjectData): void {
+		if (this.argv.release && this.argv.hmr) {
+			this.$errors.failWithoutHelp("The options --release and --hmr cannot be used simultaneously.");
+		}
+
+		// HACK: temporary solution for 5.3.0 release (until the webpack only feature)
+		const parsed = require("yargs-parser")(process.argv.slice(2), { 'boolean-negation': false });
+		const noBundle = parsed && (parsed.bundle === false || parsed.bundle === 'false');
+		if (noBundle && this.argv.hmr) {
+			this.$errors.failWithoutHelp("The options --no-bundle and --hmr cannot be used simultaneously.");
+		}
+
+		if (projectData && projectData.useLegacyWorkflow === false) {
+			this.argv.bundle = this.argv.bundle !== undefined ? this.argv.bundle : "webpack";
+			this.argv.hmr = !this.argv.release;
+		}
+
+		// --no-hmr -> hmr: false or --hmr false -> hmr: 'false'
+		const noHmr = parsed && (parsed.hmr === false || parsed.hmr === 'false');
+		if (noHmr) {
+			this.argv.hmr = false;
+		}
+
+		if (noBundle) {
```

Comment:
Maybe add a warning for the user that we will override both hmr and bundle options.